File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/cpp/cwe-22/code_after1.cpp
// Source: Row 2 in ./dataset/CVEfixes/Analysis/results/C++/df_c++_cwe_22.xlsx

CString CWebSock::GetSkinPath(const CString& sSkinName) {
    const CString sSkin = sSkinName.Replace_n("/", "_").Replace_n(".", "_");

    CString sRet = CZNC::Get().GetZNCPath() + "/webskins/" + sSkin;

    if (!CFile::IsDir(sRet)) {
        sRet = CZNC::Get().GetCurPath() + "/webskins/" + sSkin;

        if (!CFile::IsDir(sRet)) {
            sRet = CString(_SKINDIR_) + "/" + sSkin;
        }
    }

    return sRet + "/";
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/python/cwe-22/code_after1.py
# Source: Row 86 in ./dataset/CVEfixes/Analysis/results/Python/df_python_cwe_22.xlsx

from werkzeug.utils import safe_join

@app.route('/images/<path:path>')
def get_image(path):
    def get_absolute_path(path):
        import os
        script_dir = os.path.dirname(__file__)  # <-- absolute dir the script is in
        rel_path = path
        abs_file_path = safe_join(script_dir, rel_path)
        return abs_file_path

    return send_file(
        get_absolute_path(f"./images/{path}"),
        mimetype='image/png',
        attachment_filename='snapshot.png',
        cache_timeout=0
    )
